### PR TITLE
Feat/bubble series legend

### PR DESCRIPTION
**Pull request by dariuszlacheta**

### CODE REVIEW

File: packages/widgets/src/components/bubble/bubble-chart.widget.tsx [hunk 1]
```diff
@@ -4,9 +4,14 @@ import {
   BubbleChartSettings,
   ResponsiveWrapper,
   BubbleLegend,
+  SeriesLegend,
   theme as defaultTheme,
 } from '@keen.io/charts';
 
+import { scaleOrdinal } from 'd3-scale';
```
Comment (maciejrybaniec) [inline packages/widgets/src/components/bubble/bubble-chart.widget.tsx:11]: The `keen.io/widgets` do not have `d3-scale` dependency. Please move it to `keen.io/charts` and export helper function like `createColorScale`.

File: packages/widgets/src/components/bubble/bubble-chart.widget.utils.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const createLegendLabels = (
```
Comment (maciejrybaniec) [inline packages/widgets/src/components/bubble/bubble-chart.widget.utils.ts:1]: Please, add a unit tests for that utility function.

File: packages/widgets/src/types.ts [hunk 3]
```diff
@@ -32,7 +32,12 @@ export type WidgetSettings = {
   /** Widget subtitle */
   subtitle: TextSettings;
   /** Legend component settings */
-  legend?: LegendSettings;
+  legend?: LegendSettings & BubbleWidgetLegendSettings;
```
Comment (maciejrybaniec) [inline packages/widgets/src/types.ts:35]: The `&` stands for type union. We need to use alternative here so the legend types should look like:
```
interface LegendSettings {}

interface BubbleWidgetLegendSettings extends LegendSettings { ... }

legend?: LegendSettings | BubbleWidgetLegendSettings;

```

Comment (maciejrybaniec) [inline packages/widgets/src/types.ts:35]: As in me previous comment it should be alternative - not union - `LegendSettings | BubbleWidgetLegendSettings `

Comment (dariuszlacheta) [reply packages/widgets/src/types.ts]: Sorry for that, now should be corrected

File: packages/dataviz/src/render-widget.tsx [hunk 4]
```diff
@@ -45,6 +46,7 @@ type Options = {
   scaleSettings: Partial<ScaleSettings>;
   componentSettings?: ComponentSettings;
   widgetSettings?: WidgetSettings;
+  legend?: LegendSettings;
```
Comment (maciejrybaniec) [inline packages/dataviz/src/render-widget.tsx:49]: Based on that definition - the renderWidget function could receive only generic `LegendSettings`. But it's not true as `BubbleLegendSettings` could also be provided as argument. Please cover that case here.

**Comment by maciejrybaniec** [comment]
Container with legend should have defined max-height

**Pull request merged**